### PR TITLE
chore: Refactor StatusIndicator component

### PR DESCRIPTION
**Pull request by harshanarisetty**
## Summary

Fixes: #2392 

StatusIndicator component now uses Canvas Tokens and new styling utilities. Components now supports the `cs` prop, but otherwise the API has not changed. It should behave identically as it did in previous versions.

## Release Category
<a href="https://camo.githubusercontent.com/1551452433f17e7cb88cbb052252a70d73e7078ce660fdf88d5285443fc0ac6a/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f72656c656173655f63617465676f72792d436f6d706f6e656e74732d626c7565" rel="nofollow">category</a>
### Release Note
`StatusIndicator` now uses [Canvas Tokens](https://workday.github.io/canvas-tokens/?path=/docs/docs-getting-started--docs) and our [new styling utilities](https://workday.github.io/canvas-kit/?path=/docs/styling-basics--create-modifiers#createstyles-api).
The component now supports the `cs` prop, but otherwise the API has not changed. It should behave
identically as it did in previous versions.

---

## Checklist

- [ ] MDX documentation adheres to Canvas Kit's [Documentation Guidelines](https://workday.github.io/canvas-kit/?path=/docs/guides-documentation-guidelines--page)
- [ ] Label `ready for review` has been added to PR

## For the Reviewer

- [ ] PR title is short and descriptive
- [ ] PR summary describes the change (Fixes/Resolves linked correctly)
- [ ] PR Release Notes describes additional information useful to call out in a release message or removed if not applicable
- [ ] Breaking Changes provides useful information to upgrade to this code or removed if not applicable

## Areas for Feedback? (optional)

- [ ] Code
- [ ] Documentation
- [ ] Testing
- [ ] Codemods

### CODE REVIEW

**Review by RayRedGoose — CHANGES_REQUESTED** [review]
Stencil allows to set variables and text and icon use the same color. Background icon color doesn't have any sense to be used, because it's transparent by default. 

You can define styles for the icon inside stencil like this:

```ts
const statusIndicatorStencil = createStencil({
  vars: {
    // set variable, we can specify a default color as `base.licorice400` 
    // or keep it empty because it's managed by modifiers
    color: '',
  },
  base: ({color}) => ({
    ...
    // specify that for text and icon colors we use stencil variable
    color: color,
    '& .wd-icon-fill': {
      fill: color,
    },
  }),
  modifiers: {
    gray: {
      // just reassigning the `color` variable
      high: ({color}) => ({
        [color]: base.frenchVanilla100,
        background: base.licorice300,
      }),
      // just reassigning the `color` variable
      low: ({color}) => ({
        [color]: base.licorice400,
        background: base.soap300,
      }),
    },
  },
  // also set default modifier as `gray-low`, so it helps us remove model easily
  defaultModifiers: {
    gray: 'low',
  },
}
```

With this stencil we doesn't need to set colors for `StatusIndicatorIcon` and model can be removed for this component.

File: modules/preview-react/status-indicator/lib/StatusIndicator.tsx [hunk 1]
```diff
@@ -13,13 +16,92 @@ export interface StatusIndicatorProps extends ExtractProps<typeof Flex, never> {
...
+        background: base.blackPepper600,
+      },
+      low: {
+        opacity: 0.85,
+        color: base.frenchVanilla100,
+        [systemIconStencil.vars.color]: base.frenchVanilla100,
+        background: base.blackPepper600,
+      },
+    },
+  },
```
Comment (RayRedGoose) [inline modules/preview-react/status-indicator/lib/StatusIndicator.tsx:104]: We will need to have default modifier too. It's handled by model now, but model will be removed in a future.

**Review by RayRedGoose — COMMENTED** [review]
Everything looks good! But `StatusIndicatorLabel` should be updated too and style props should be removed.

File: modules/preview-react/status-indicator/lib/StatusIndicator.tsx (hunk 1 above)
Comment (harshanarisetty) [reply modules/preview-react/status-indicator/lib/StatusIndicator.tsx:104]: Added back the default Modifier

File: modules/docs/mdx/11.0-UPGRADE-GUIDE.mdx [hunk 2]
```diff
@@ -727,6 +727,17 @@ longer extends the `Box` component, however, it still supports Emotion `styled`
...
+### StatusIndicator (Preview)
+
+**PR:** [#2620] https://github.com/Workday/canvas-kit/pull/2620
+
+`StatusIndicator` now uses
+[Canvas Tokens](https://workday.github.io/canvas-tokens/?path=/docs/docs-getting-started--docs) and
+our
+[new styling utilities](https://workday.github.io/canvas-kit/?path=/docs/styling-basics--create-modifiers#createstyles-api).
+The component now supports the `cs` prop, but otherwise the API has not changed. It should behave
+identically as it did in previous versions.
```
Comment (mannycarrera4) [inline modules/docs/mdx/11.0-UPGRADE-GUIDE.mdx:739]: I think a breaking change is that we've removed `useStatusIndicatorIcon` which was once exported. We should document this.

File: modules/preview-react/status-indicator/lib/StatusIndicator.tsx [hunk 3]
```diff
@@ -13,13 +16,95 @@ export interface StatusIndicatorProps extends ExtractProps<typeof Flex, never> {
   children: React.ReactNode;
 }
 
-export const useStatusIndicator = createElemPropsHook(useStatusIndicatorModel)(({state}) => {
-  const colors = statusIndicatorColors[state.variant][state.emphasis];
-  return {
-    opacity: state.variant === 'transparent' ? '0.85' : undefined,
-    color: colors.text,
-    background: colors.background,
-  };
```
Comment (mannycarrera4) [inline modules/preview-react/status-indicator/lib/StatusIndicator.tsx:22]: This is a breaking change that we should add to the upgrade guide

Comment (RayRedGoose) [reply modules/preview-react/status-indicator/lib/StatusIndicator.tsx:22]: Oh, yeah. It is. I'm surprised it's exported.

File: modules/docs/mdx/11.0-UPGRADE-GUIDE.mdx (hunk 2 above)
Comment (harshanarisetty) [reply modules/docs/mdx/11.0-UPGRADE-GUIDE.mdx:739]: I added them back, can you please take a look?

File: modules/preview-react/status-indicator/lib/StatusIndicator.tsx (hunk 3 above)
Comment (harshanarisetty) [reply modules/preview-react/status-indicator/lib/StatusIndicator.tsx:22]: I added the back, can you please take a look?

**Review by RayRedGoose — APPROVED** [review]

**Pull request merged**